### PR TITLE
Provide quickstart guide

### LINKED ISSUE #45: Provide a quick-start guide in the README
# Improve documentation

## Link 

[README](https://github.com/supabase/storage-js/blob/main/README.md)

## Describe the problem

In other supabase components, such as [postgrest-js](https://github.com/supabase/postgrest-js/blob/master/README.md#quick-start), [realtime-js](https://github.com/supabase/realtime-js#usage) and [gotrue-js](https://github.com/supabase/gotrue-js#quick-start), there are short quick start sections in the readme to show you how you can get up and running without using the `supabase-js` package.

## Describe the improvement

Add a few code snippets that will document a quick start procedure.

### PR DESCRIPTION
**Pull request by mirceanton**
## What kind of change does this PR introduce?

Provided a quickstart guide in the readme to showcase how one could get up and running using this module by itself, not as part of `supabase-js`

Fixes #45

### CODE REVIEW

**Comment by mikeanth-dev** [comment]
Note. Reviewing PR #48 is a prerequisite, as the examples will slightly change if that is accepted.

File: README.md [hunk 1]
```diff
@@ -4,6 +4,77 @@ JS Client library to interact with Supabase Storage.
...
+``` js
+// index.hs
+import { SupabaseStorageClient } from '@supabase/storage-js'
+import 'dotenv/config'
+
+const STORAGE_URL = process.env.SUPABASE_STORAGE_URL;
+const SERVICE_KEY = process.env.SUPABASE_SERVICE_KEY; //! service key, not anon key
+
+const storage_client = new SupabaseStorageClient(
+    STORAGE_URL,
```
Comment (thebengeu) [inline README.md:32]: Nit: indentation of 2 spaces for consistency with documentation elsewhere?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,77 @@ JS Client library to interact with Supabase Storage.
...
+``` bash
+# .env
+SUPABASE_STORAGE_URL=https://<project_ref>.supabase.co/storage/v1
+SUPABASE_SERVICE_KEY=<service_role>
+```
+
+``` js
+// index.hs
+import { SupabaseStorageClient } from '@supabase/storage-js'
+import 'dotenv/config'
```
Comment (thebengeu) [inline README.md:26]: Not everyone would use `dotenv`. So I think just having the constants directly, without using environment variables, would be good enough for an example.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,77 @@ JS Client library to interact with Supabase Storage.
...
+
+const STORAGE_URL = process.env.SUPABASE_STORAGE_URL;
+const SERVICE_KEY = process.env.SUPABASE_SERVICE_KEY; //! service key, not anon key
+
+const storage_client = new SupabaseStorageClient(
```
Comment (thebengeu) [inline README.md:31]: camelCase for consistency with documentation elsewhere? Here and below.

File: README.md [hunk 4]
```diff
@@ -4,6 +4,77 @@ JS Client library to interact with Supabase Storage.
...
+    const { data, error } = await storage_client.createBucket("test_bucket", { public: false });
+    ```
+
+- Retrieve the details of an existing Storage bucke
+
+    ``` js
+    const { data, error } = await storage_client.getBucket("test_bucket");
+    ```
+
+- Updates a new Storage bucket
```
Comment (thebengeu) [inline README.md:54]: The operation in the example doesn't match.

File: README.md [hunk 5]
```diff
@@ -4,6 +4,77 @@ JS Client library to interact with Supabase Storage.
...
+
+    ``` js
+    const { data, error } = await storage_client.createBucket("test_bucket", { public: true });
+    ```
+
+- Remove all objects inside a single bucket
```
Comment (thebengeu) [inline README.md:60]: The operation in the example doesn't match.

**Review by thebengeu — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: README.md (hunk 1 above)
Comment (mikeanth-dev) [reply README.md:32]: Will solve this tonight when i get home!

File: README.md (hunk 3 above)
Comment (mikeanth-dev) [reply README.md:31]: Uh oh. Bad copy paste job. My bad :))

Will fix

File: README.md (hunk 5 above)
Comment (mikeanth-dev) [reply README.md:58]: Fixed in [this commit](https://github.com/supabase/storage-js/pull/47/commits/d6cf0313a897b738be9d292dcc349571cf81aec0)

File: README.md (hunk 4 above)
Comment (mikeanth-dev) [reply README.md:49]: Fixed in [this commit](https://github.com/supabase/storage-js/pull/47/commits/d6cf0313a897b738be9d292dcc349571cf81aec0)

File: README.md (hunk 2 above)
Comment (mikeanth-dev) [reply README.md]: Fixed [here](https://github.com/supabase/storage-js/pull/47/commits/f08df8322bf9b89e5b51e2465ea3a8230907633b)

File: README.md (hunk 1 above)
Comment (mikeanth-dev) [reply README.md]: [Done](https://github.com/supabase/storage-js/pull/47/commits/2aa63b7bc3624b691912300111a7028f00dd2c96)

File: README.md (hunk 3 above)
Comment (mikeanth-dev) [reply README.md]: [Fixed](https://github.com/supabase/storage-js/commit/0af5169a884d5d9f0bd30be8d79526f263c46d0e)

File: README.md [hunk 6]
```diff
@@ -4,6 +4,75 @@ JS Client library to interact with Supabase Storage.
...
+  const { data, error } = await storageClient.emptyBucket("test_bucket");
+  ```
+
+- Deletes an existing bucket (a bucket can't be deleted with existing objects inside it):
+
+  ``` js
+  const { data, error } = await storageClient.deleteBucket("test_bucket");
+  ```
+
+- Retrieve the details of all Storage buckets within an existing product:
```
Comment (thebengeu) [inline README.md:70]: For consistency, either end all the list items with a colon, or don't? Personally I'd go with the colons.

Comment (thebengeu) [inline README.md:70]: "project" instead of "product"?

**Review by thebengeu — APPROVED** [review]
Looks great! Two minor nits.

File: README.md (hunk 6 above)
Comment (mikeanth-dev) [reply README.md:70]: In that case, might I suggest I also change the comment [here](https://github.com/mikeanth-dev/storage-js/blob/0af5169a884d5d9f0bd30be8d79526f263c46d0e/src/lib/StorageBucketApi.ts#L18)?

The action descriptions have been copy-pasted from the comments in the code.

Comment (thebengeu) [reply README.md:70]: Yes, do change that comment.

Comment (mikeanth-dev) [reply README.md]: fixed.

Comment (mikeanth-dev) [reply README.md]: agreed and fixed.

**Comment by mikeanth-dev** [comment]
I was also thinking whether or not [this line](https://github.com/mikeanth-dev/storage-js/blob/b6b5acfc2876daaf2973c4c9815b8b144c0f90bb/README.md?plain=1#L32) should be changed to `Handling storage buckets` or just `Handling buckets` and then add another section to show basic functionality with files within buckets. Actions like:

- Uploading a file
- Downloading a file
- Removing a file
- Moving/Copying a file
- Updating a file

Is that taking it too far for a quickstart guide? I think it could be useful to have those examples as well.

**Comment by thebengeu** [comment]
`Handling storage buckets` sounds good to me.

> Is that taking it too far for a quickstart guide?  I think it could be useful to have those examples as well.

I agree that it could be useful. What you've done is already a great start, I'll leave it to you whether you want to add more examples.

**Comment by mikeanth-dev** [comment]
I will add more examples after #48 gets merged, so that I can update them too to use `StorageClient` instead.

**Comment by mirceanton** [comment]
Converted this PR to a draft so it doesn't get prematurely merged since I am not done working on it. I will add more examples, hopefully by the end of this week. I've been busy with work and other projects.

**Comment by mirceanton** [comment]
@thebengeu maybe something like this?

File: README.md [hunk 7]
```diff
@@ -4,6 +4,139 @@ JS Client library to interact with Supabase Storage.
...
+  ```
+
+  > Note: The `list` method also accepts a map of optional parameters. For a complete list see the [Supabase API reference](https://supabase.com/docs/reference/javascript/storage-from-list).
+
+- Replace an existing file at the specified path with a new one:
+
+  ```js
+  const fileBody = ...; // load your file here
+  const { data, error } = await storageClient.from('bucket')
+  .update('path/to/file', fileBody);
```
[the remark this answers is not in the document]
Comment (mirceanton) [reply README.md]: [Fixed](https://github.com/supabase/storage-js/pull/47/commits/3159b838dc7f9e167b54bdf5a42aaf543d32dd77).

File: README.md [hunk 8]
```diff
@@ -4,6 +4,139 @@ JS Client library to interact with Supabase Storage.
...
+  ```
+
+#### Handling Files
+
+- Upload a file to an existing bucket:
+
+  ```js
+  const fileBody = ...; // load your file here
+
+  const { data, error } = await storageClient.from('bucket').upload('path/to/file', fileBody);
```
[the remark this answers is not in the document]
Comment (mirceanton) [reply README.md]: [Fixed](https://github.com/supabase/storage-js/pull/47/commits/3159b838dc7f9e167b54bdf5a42aaf543d32dd77).

**Pull request merged**